### PR TITLE
Upgrade AGP 7.3.0

### PR DESCRIPTION
**Pull request by ivaniskandar**
Removed the pinned R8 version as it doesn't work with this AGP. So this may cause some issues with extensions.

### CODE REVIEW

**Comment by arkon** [comment]
Did you test with R8?

**Comment by ivaniskandar** [comment]
Yeah I've used this on my personal build for a while

File: app/proguard-rules.pro [hunk 1]
```diff
@@ -13,6 +12,14 @@
 -keep,allowoptimization class app.cash.quickjs.** { public protected *; }
 -keep,allowoptimization class uy.kohesive.injekt.** { public protected *; }
 
+# From extensions-lib
```
Comment (arkon) [inline app/proguard-rules.pro:15]: Might be more convenient to just annotate things with `@Keep`

File: source-api/consumer-proguard.pro [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-keep,allowoptimization class eu.kanade.tachiyomi.source.model.** { public protected *; }
```
Comment (arkon) [inline source-api/consumer-proguard.pro:1]: Wouldn't we realistically want to keep everything in this module?

Comment (ivaniskandar) [reply source-api/consumer-proguard.pro:1]: Yeah but the `eu.kanade.tachiyomi.source` package also contains some stuff from main app module

File: app/proguard-rules.pro (hunk 1 above)
Comment (ivaniskandar) [reply app/proguard-rules.pro:15]: Honestly I prefer keeping the rules centralized but I can try using it.

Comment (arkon) [reply app/proguard-rules.pro:15]: I'll merge this one first to keep the changes somewhat isolated.

**Pull request merged**